cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -607,6 +607,29 @@ TEST_P(TestParquetFileFormatScan, PredicatePushdown) {
                             kNumRowGroups - 5);
 }
 
+TEST_P(TestParquetFileFormatScan, RangeScan) {
+  constexpr int64_t kNumRowGroups = 16;
+  constexpr int64_t kTotalNumRows = kNumRowGroups * (kNumRowGroups + 1) / 2;
```

Comment:
any specific reason for this exact number?

Reply:
following existing tests

Reply:
Noted.